cpp/benchmarks/common/random_distribution_factory.hpp

```diff
@@ -18,17 +18,34 @@
 
 #include "generate_input.hpp"
 
+#include <cmath>
+#include <cstddef>
 #include <memory>
 #include <random>
+#include <rmm/device_uvector.hpp>
+#include <type_traits>
 
+#include <thrust/execution_policy.h>
+#include <thrust/random.h>
+#include <thrust/random/normal_distribution.h>
+#include <thrust/random/uniform_int_distribution.h>
+#include <thrust/tabulate.h>
 /**
  * @brief Generates a normal(binomial) distribution between zero and upper_bound.
  */
-template <typename T, typename std::enable_if_t<std::is_integral<T>::value, T>* = nullptr>
+template <typename T,
+          typename std::enable_if_t<std::is_integral<T>::value or std::is_same_v<__int128_t, T>,
```

Comment:
```suggestion
          typename std::enable_if_t<cuda::std::is_integral<T>::value>,
```

Reply:
Should we do this for entire cudf also at once?